server.py

```diff
@@ -159,6 +159,25 @@ def delete_prod_info(prod_id):
         prod_info.delete()
     return make_response('Product information deleted.', status.HTTP_200_OK)
 
+@app.route(PATH_INVENTORY_PROD_ID, methods=[PUT])
+def update_prod_info(prod_id):
+    """
+    Update prodcut information
+
+    This endpoint will update a ProductInformation based on data that id posted in the body
+    """
+    check_content_type('application/json')
+    prod_info = ProductInformation.find(prod_id)
+    if not prod_info:
+        raise NotFound("Product with id '{}' was not found.".format(prod_id))
+
+    prod_info.deserialize_update(request.get_json())
+    if prod_info.prod_id == prod_id:
```

Comment:
This if-else block is redundant since prod_info.prod_id comes from ProructInformation.find(prod_id). Also, your deserialize_update() doesn't take care of `prod_id`. This condition check must be True.
If you are worrying about ProductInformation.find() not returning the correct prod_info, add a test for models.py's find() instead.